// jest-globals/src/Location.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMockLocation = void 0;
const jsuri_1 = __importDefault(require("jsuri"));
const mocks_1 = require("./mocks");
const createMockLocation = () => {
    const mockLocation = {
        assign: jest.fn((url) => {
            const currentUri = new jsuri_1.default(window.location.href);
            const newUri = new jsuri_1.default(url);
            if (!newUri.host()) {
                newUri.setProtocol(currentUri.protocol());
                newUri.setHost(currentUri.host());
            }
            mockLocation.href = newUri.toString();
            mockLocation.host = newUri.host();
            mockLocation.origin = `${newUri.protocol()}://${newUri.host()}`;
            mockLocation.pathname = newUri.path();
            mockLocation.search = newUri.query();
        }),
        hash: "",
        href: "",
        host: "",
        origin: "",
        pathname: "",
        search: "",
        reload: (0, mocks_1.fn)(),
    };
    return mockLocation;
};
exports.createMockLocation = createMockLocation;
