// jest-globals/src/Window.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MockWindow = void 0;
const Location_1 = require("./Location");
const MediaQueryList_1 = require("./MediaQueryList");
const mocks_1 = require("./mocks");
const Navigator_1 = require("./Navigator");
const Performance_1 = require("./Performance");
const Storage_1 = require("./Storage");
class MockWindow {
    constructor(settings = {}) {
        var _a;
        this.addEventListener = (0, mocks_1.fn)();
        this.alert = (0, mocks_1.fn)();
        this.blur = (0, mocks_1.fn)();
        this.close = (0, mocks_1.fn)();
        this.confirm = (0, mocks_1.fn)();
        this.dispatchEvent = (0, mocks_1.fn)();
        this.focus = (0, mocks_1.fn)();
        this.getSelection = (0, mocks_1.fn)();
        this.localStorage = new Storage_1.MockStorage();
        this.location = (0, Location_1.createMockLocation)();
        this.matchMedia = (0, mocks_1.fn)().mockReturnValue((0, MediaQueryList_1.createMockMediaQueryList)());
        this.moveBy = (0, mocks_1.fn)();
        this.navigator = (0, Navigator_1.createMockNavigator)();
        this.open = (0, mocks_1.fn)();
        this.performance = (0, Performance_1.createMockPerformance)();
        this.postMessage = (0, mocks_1.fn)();
        this.print = (0, mocks_1.fn)();
        this.prompt = (0, mocks_1.fn)();
        this.removeEventListener = (0, mocks_1.fn)();
        this.requestAnimationFrame = (0, mocks_1.fn)();
        this.requestIdleCallback = (0, mocks_1.fn)();
        this.resizeBy = (0, mocks_1.fn)();
        this.resizeTo = (0, mocks_1.fn)();
        this.scrollBy = (0, mocks_1.fn)();
        this.scrollTo = (0, mocks_1.fn)();
        this.sessionStorage = new Storage_1.MockStorage();
        this.setImmediate = (0, mocks_1.fn)();
        this.stop = (0, mocks_1.fn)();
        Object.assign(this, settings);
        (_a = this.top) !== null && _a !== void 0 ? _a : (this.top = this);
    }
}
exports.MockWindow = MockWindow;
